tests/View/Blade/BladeBoolTest.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Illuminate\Tests\View\Blade;
+
+class BladeBoolTest extends AbstractBladeTestCase
+{
+    public function testBool()
```

Comment:
This is probably one of the few cases where an `eval()` test would have found the error above, like so:

```php
public function testCompileBool(): void
    {
        $someViewVarTruthy = 123;
        $compiled = $this->compiler->compileString('@bool($someViewVarTruthy)');

        ob_start();
        eval(substr($compiled, 6, -3));
        $this->assertEquals('true', ob_get_clean());

        $someViewVarFalsey = '0';
        $compiled = $this->compiler->compileString('@bool($someViewVarFalsey)');

        ob_start();
        eval(substr($compiled, 6, -3));
        $this->assertEquals('false', ob_get_clean());

    }
```

Reply:
You're right, it didn't consider a Falsey values, i added a Falsey support and a couple of test for NULL and Class instances.